api/main.go

```diff
@@ -73,112 +73,25 @@ func PricePlanFile(ctx context.Context, jsonBlob string) (float64, []string, []s
 		return 0.0, []string{}, []string{}, err
 	}
 	var hourlyPrice float64
-	var resources []common.Priceable
+	pricer := PlanfilePricerInstance(ctx)
+	pricer.RegisterAssetPricer(azure.VirtualMachineAssetPricer{})
```

Comment:
Could the `VirtualMachine` type simply implement this interface and we don't need an additional wrapper pricer? Types can implement many interfaces.